src/contracts/security/security.service.ts

```diff
@@ -151,28 +179,175 @@ export class SecurityService {
-    this.logger.warn('Waiting for block confirmation');
+    this.logger.warn('Pause transaction sent', {
+      txHash: tx.hash,
+      blockNumber,
+      stakingModuleId,
+    });
+    this.logger.warn('Waiting for block confirmation', {
+      blockNumber,
+      stakingModuleId,
+    });
+
+    await tx.wait();
+
+    this.logger.warn('Block confirmation received', {
+      blockNumber,
+      stakingModuleId,
+    });
+  }
+
+  /**
+   * Signs a message to unvet keys
+   */
+  public async signUnvetData(
+    nonce: number,
+    blockNumber: number,
+    blockHash: string,
+    stakingModuleId: number,
+    operatorIds: string,
+    vettedKeysByOperator: string,
+  ): Promise<Signature> {
+    const prefix = await this.repositoryService.getUnvetMessagePrefix();
+
+    return await this.walletService.signUnvetData({
+      prefix,
+      blockNumber,
+      blockHash,
+      stakingModuleId,
+      nonce,
+      operatorIds,
+      vettedKeysByOperator,
+    });
+  }
+
+  /**
+   * Send transaction to unvet signing keys
+   */
+  @OneAtTime()
+  public async unvetSigningKeys(
+    nonce: number,
+    blockNumber: number,
+    blockHash: string,
+    @StakingModuleId stakingModuleId: number,
+    operatorIds: string,
+    vettedKeysByOperator: string,
+    signature: Signature,
+  ): Promise<ContractReceipt | void> {
+    this.logger.warn('Try to unvet keys for staking module', {
+      stakingModuleId,
+      blockNumber,
+    });
+
+    const contract = this.getContractWithSigner();
+
+    const { r, _vs: vs } = signature;
+    const tx = await contract.unvetSigningKeys(
+      blockNumber,
+      blockHash,
+      stakingModuleId,
+      nonce,
+      operatorIds,
+      vettedKeysByOperator,
+      {
+        r,
+        vs,
+      },
+    );
+
+    this.logger.warn('Unvet transaction sent', {
+      txHash: tx.hash,
+      blockNumber,
+      stakingModuleId,
+    });
+    this.logger.warn('Waiting for block confirmation', {
+      blockNumber,
+      stakingModuleId,
+    });
 
     await tx.wait();
 
-    this.logger.warn('Block confirmation received');
+    this.logger.warn('Block confirmation received', {
+      blockNumber,
+      stakingModuleId,
+    });
+  }
+
+  /**
+   * Return the maximum number of operators in one unvetting transaction
+   */
+  public async getMaxOperatorsPerUnvetting(
+    blockTag?: BlockTag,
+  ): Promise<number> {
+    const contract = this.getContractWithSigner();
+
+    const maxOperatorsPerUnvetting = await contract.getMaxOperatorsPerUnvetting(
+      {
+        blockTag: blockTag as any,
+      },
+    );
+
+    return maxOperatorsPerUnvetting.toNumber();
+  }
+
+  public async version(blockTag?: BlockTag): Promise<number> {
+    const contract = this.getContractWithSigner();
+    try {
+      const version = await contract.VERSION({
+        blockTag: blockTag as any,
+      });
+      return version.toNumber();
+    } catch (error) {
```

Comment:
I think we need to somehow handle the case if there is an error not related to the fact that there is no such method. if we switch back to the first version in case of an error, there may be errors

Reply:
from this error is not possible to identify it. but it is a temporary solution.  in v3 contract version is a constant.
we will not ser version 1 as a default.